packages/flutter/lib/src/painting/text_scaler.dart

```diff
@@ -140,13 +140,17 @@ final class _ClampedTextScaler implements TextScaler {
     if (identical(this, other)) {
       return true;
     }
+    assert(minScale < maxScale);
     return other is _ClampedTextScaler &&
         minScale == other.minScale &&
         maxScale == other.maxScale &&
-        (minScale == maxScale || scaler == other.scaler);
+        scaler == other.scaler;
   }
 
   @override
   int get hashCode =>
       minScale == maxScale ? minScale.hashCode : Object.hash(scaler, minScale, maxScale);
+
+  @override
+  String toString() => '$scaler clamped [$minScale, $maxScale]';
```

Comment:
consider use Diagnosticable

Reply:
That doesn't seem to reduce the amount of code I have to write, are there other benefits of using `Diagnosticable`?

Reply:
it provide a better api to fill in properties for printing the string representation. Besides that, I think devtools uses that api to get details about an object, but i am not too familiar with that part of the code.

Reply:
This is what it looks like now: 
```
'│└RichText(softWrap: wrapping at box width, textScaler: SystemTextScaler (no scaling), maxLines: unlimited, text: "A", dependencies: [Directionality], renderObject: RenderParagraph#00000 relayoutBoundary=up1)\n'
```

So that still works. I'd keep the class hierarchy simple for now in case someone wants to `implement` the class instead of extending.

Reply:
yeah makes sense